### ISSUE TITLE
How to run the same time under different refresh rates

### ISSUE THREAD

**Issue by gygy1993** [dimforge/rapier#483]
For example, apply a force to the ball to make it roll. Run it at 60fps and 144fps. 144fps takes a lot less time than 60fps. js version

**Comment by isaac-mason** [comment]
You might want fixed time steps with interpolation.

This is a good post on the topic: https://gafferongames.com/post/fix_your_timestep/

**Comment by isaac-mason** [comment]
A good reference for doing this with the js lib could be the @react-three/rapier library: https://github.com/pmndrs/react-three-rapier

See: https://github.com/pmndrs/react-three-rapier/blob/main/packages/react-three-rapier/src/components/Physics.tsx#L395

It's a react library but the approach isn't specific to react.

**Comment by gygy1993** [comment]
> A good reference for doing this with the js lib could be the `@react-three/rapier` library: https://github.com/pmndrs/react-three-rapier
> 
> See: https://github.com/pmndrs/react-three-rapier/blob/main/packages/react-three-rapier/src/components/Physics.tsx#L395
> 
> It's a react library but the approach isn't specific to react.
> 
> If you've got more questions specific to the js library though, probably best to create a discussion on the rapier.js GitHub repo, or post some questions in the discord.

Thanks for your answer, see timestep parameter in world

**Comment by isaac-mason** [comment]
Yep changing the timestep each frame will be variable time stepping, which is good for many use cases, but has tradeoffs with determinism.
